service-base/src/main/java/org/eclipse/hono/service/NewAbstractApplication.java

```diff
@@ -71,15 +71,17 @@ protected Vertx getVertx() {
     }
 
     /**
-     * Adds multiple serviceFactories to this server.
+     * Sets the factories to use for creating service instances to deploy the Vert.x container during startup.
+     * <p>
      *
-     * @param definedServiceFactories The serviceFactories.
+     * @param factories The service factories.
+     * @throws NullPointerException if factories is {@code null}.
      */
-    @Autowired(required = false)
-    public final void addServiceFactories(final Collection<ObjectFactory<? extends AbstractServiceBase>> definedServiceFactories) {
-        Objects.requireNonNull(definedServiceFactories);
-        serviceFactories.addAll(definedServiceFactories);
-        log.debug("added {} service factories", definedServiceFactories.size());
+    @Autowired
+    public final void setServiceFactories(final Set<ObjectFactory<? extends AbstractServiceBase>> factories) {
+        Objects.requireNonNull(factories);
+        serviceFactories = factories;
```

Comment:
Ah, now I see what you mean. But the problem remains: the method's name says setter but in fact it adds elements. Of course calling the setter twice might be an edge case, but still irritating. We could empty the set before invoking `serviceFactories.addAll(factories)` or rename the method back to `addServiceFactories`.